core/controllers/admin.py

```diff
@@ -854,3 +854,37 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+
+        if new_username is None or current_username is None:
+            raise self.InvalidInputException(
+                'Please ensure that the usernames are not empty.')
+
+        if type(current_username).__name__ != 'unicode' or type(
+                new_username).__name__ != 'unicode':
+            raise self.InvalidInputException(
+                'Please ensure that the usernames are strings.')
```

Comment:
`Expected current_username to be a string, received %s` (Have a separate check!)

Reply:
Done, thanks!